### PR TITLE
Task1 complete

### PR DESCRIPTION
**Pull request by AliokY**
Task1 completed

### CODE REVIEW

**Comment by AliokY** [comment]
HW_15

File: HW_15/HW15.Task3/Attributes/CreditCardAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            Match match = regex.Match(value.ToString());
+
+            if (match.Success)
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
```
Comment (VadzimPapko) [inline HW_15/HW15.Task3/Attributes/CreditCardAttribute.cs:30]: just simplify it -` return match.Sucess`

File: HW_15/HW15.Task3/Attributes/CreditCardAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.Text.RegularExpressions;
+
+namespace HW15.Task3
+{
+    [AttributeUsage(AttributeTargets.Property)]
+    class CreditCardAttribute : ValidationAttribute
```
Comment (VadzimPapko) [inline HW_15/HW15.Task3/Attributes/CreditCardAttribute.cs:8]: good one

File: HW_15/HW15.Task3/Program.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            //    { Console.WriteLine("Credit card is accepted"); }
+            //    else
+            //    { Console.WriteLine("Credit card isn't accepted"); }
+            //}
+
+            ChangeConst();
+
+            Console.ReadKey();
+        }
+        static bool CheckCreditCard(string creditCardNumber)
```
Comment (VadzimPapko) [inline HW_15/HW15.Task3/Program.cs:34]: good one

File: HW_15/HW15/Attributes/CreditCardAttribute.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            Match match = regex.Match(value.ToString());
+
+            if (match.Success)
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
```
Comment (VadzimPapko) [inline HW_15/HW15/Attributes/CreditCardAttribute.cs:32]: take a look at my comment above

File: HW_15/Motoshop/Motoshop/Attributes/GuidAttribute.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        {
+            if (value is null)
+            {
+                return false;
+            }
+
+            bool parseResult = Guid.TryParse(value.ToString(), out Guid _);
+
+            return parseResult;
+        }
```
Comment (VadzimPapko) [inline HW_15/Motoshop/Motoshop/Attributes/GuidAttribute.cs:19]: good one

File: HW_15/Motoshop/Motoshop/Attributes/MinYearAttribute.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+namespace Motoshop.Attributes
+{
+    public class MinYearAttribute : ValidationAttribute
+    {
+        public int MinYear { get; }
+        public MinYearAttribute(int minYear)
+        {
+            MinYear = minYear;
+        }
+        public string GetErrorMessage() => $"Motorcycle must be made no early than {MinYear}";
```
Comment (VadzimPapko) [inline HW_15/Motoshop/Motoshop/Attributes/MinYearAttribute.cs:17]: replace to 
image

File: HW_15/Motoshop/Motoshop/Attributes/MinYearAttribute.cs [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            var yearOfIssue = ((DateTime)value).Year;
+
+            if (MinYear <= yearOfIssue && yearOfIssue <= DateTime.Now.Year)
+            {
+                return new ValidationResult(GetErrorMessage());
```
Comment (VadzimPapko) [inline HW_15/Motoshop/Motoshop/Attributes/MinYearAttribute.cs:25]: image

**Review by VadzimPapko — COMMENTED** [review]
Reviewed. Your score is 9.5

**Pull request merged**